x-pack/plugins/security_solution/public/common/components/visualization_actions/index.test.tsx

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import { dnsTopDomainsAttrs } from './configs/network/dns_top_domains';
+import { VisualizationActions } from '.';
+import {
+  createSecuritySolutionStorageMock,
+  kibanaObservable,
+  mockGlobalState,
+  SUB_PLUGINS_REDUCER,
+  TestProviders,
+} from '../../mock';
+import { createStore, State } from '../../store';
+import { UpdateQueryParams, upsertQuery } from '../../store/inputs/helpers';
+import { cloneDeep } from 'lodash';
+import { useKibana } from '../../lib/kibana/kibana_react';
+import { CASES_FEATURE_ID } from '../../../../common/constants';
+import { mockCasesContext } from '../../../../../cases/public/mocks/mock_cases_context';
+jest.mock('react-router-dom', () => {
+  const actual = jest.requireActual('react-router-dom');
+  return {
+    ...actual,
+    useLocation: jest.fn(() => {
+      return { pathname: 'network' };
+    }),
+  };
+});
+jest.mock('../../lib/kibana/kibana_react');
+jest.mock('../../utils/route/use_route_spy', () => {
+  return {
+    useRouteSpy: jest.fn(() => [{ pageName: 'network', detailName: '', tabName: 'dns' }]),
+  };
+});
+describe('VisualizationActions', () => {
+  const refetch = jest.fn();
+  const state: State = mockGlobalState;
+  const { storage } = createSecuritySolutionStorageMock();
+  const newQuery: UpdateQueryParams = {
+    inputId: 'global',
+    id: 'networkDnsHistogramQuery',
+    inspect: {
+      dsl: ['mockDsl'],
+      response: ['mockResponse'],
+    },
+    loading: false,
+    refetch,
+    state: state.inputs,
+  };
+
+  let store = createStore(state, SUB_PLUGINS_REDUCER, kibanaObservable, storage);
+  const props = {
+    lensAttributes: dnsTopDomainsAttrs,
+    queryId: 'networkDnsHistogramQuery',
+    timerange: {
+      from: '2022-03-06T16:00:00.000Z',
+      to: '2022-03-07T15:59:59.999Z',
+    },
+    title: 'mock networkDnsHistogram',
+  };
+  const mockNavigateToPrefilledEditor = jest.fn();
+  const MockLensSaveModalComponent = jest.fn(() => (
+    <div data-test-subj="LensSaveModal">{'mockLensSaveModal'}</div>
+  ));
+  const mockGetCreateCaseFlyoutOpen = jest.fn();
+  const mockGetAllCasesSelectorModalOpen = jest.fn();
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    (useKibana as jest.Mock).mockReturnValue({
+      services: {
+        lens: {
+          canUseEditor: jest.fn(() => true),
+          navigateToPrefilledEditor: mockNavigateToPrefilledEditor,
+          SaveModalComponent: MockLensSaveModalComponent,
+        },
+        cases: {
```

Comment:
You can use the `casesMock` if you like from the cases plugin here `x-pack/plugins/cases/public/mocks.ts`